README.md

```diff
@@ -163,18 +163,25 @@ following rules:
       [Thrift definition](src/main/thrift/parquet.thrift) in the
       `ColumnOrder` union. They are summarized here but the Thrift definition
       is considered authoritative:
-      * NaNs should not be written to min or max statistics fields.
-      * If the computed max value is zero (whether negative or positive),
-        `+0.0` should be written into the max statistics field.
-      * If the computed min value is zero (whether negative or positive),
-        `-0.0` should be written into the min statistics field.
-
-      For backwards compatibility when reading files:
-      * If the min is a NaN, it should be ignored.
-      * If the max is a NaN, it should be ignored.
-      * If the min is +0, the row group may contain -0 values as well.
-      * If the max is -0, the row group may contain +0 values as well.
-      * When looking for NaN values, min and max should be ignored.
+      * The following compatibility rules should be applied when reading statistics:
+        * If the nan_count field is set to > 0 and both min and max are
```

Comment:
Personally I think https://github.com/apache/parquet-format/blob/master/src/main/thrift/parquet.thrift#L752 can together decide the status here.

Reply:
@mapleFU Yes, we could also add a `nan_pages` bool list in the column index. That would work as well.

My gut feeling is that one day having a `value_counts` count would be more useful than boolean lists. We already have `null_pages` and `null_counts` and we would then also have `nan_pages` and `nan_counts`, both `null_pages` and `nan_pages` would be obsolete if there were `value_counts`. Yes, storing one integer is likely more space than storing two booleans, but knowing the number of values in a page could also be helpful for other pruposes.

But yes, we could drop the testing of `min=max=NaN` if we had a `nan_pages` list in the column index.

Reply:
Yes, maybe you are right. My point is that, if we write nan_count or even record count, the program would works well. However, non-float point page would have some size-overhead. Personally, I'd like to use `list<bool>`, because it's easy to implement, and also lightweight. And we can hear others idea.